packages/core/src/hooks/useStarkAddress.ts

```diff
@@ -88,7 +94,7 @@ function queryFn({
     const result = await p.callContract({
       contractAddress: namingContract as string,
       entrypoint: "domain_to_address",
-      calldata: CallData.compile({ domain: encodedDomain, hint: [] }),
+      calldata: CallData.compile({ domain: encodedDomain, hint: [], chainId:{chainId} }),
```

Comment:
```suggestion
      calldata: CallData.compile({ domain: encodedDomain, hint: [] }),
```

This is not were chainId is used.

Reply:
You need to use `chainId` above in place of `network`. Change `StarknetIdNamingContract` to have the chain ids as keys and not the network names. Don't hardcode chainids, but use `@starknet-react/chains`.

```ts
import { mainnet, sepolia } from "@starknet-react/chains";

// then access the ids as follow:
mainnet.id
sepolia.id
```